src/flamenco/runtime/fd_pubkey_utils.c

```diff
@@ -31,3 +33,72 @@ fd_pubkey_create_with_seed( fd_exec_instr_ctx_t const * ctx,
 
   return FD_EXECUTOR_INSTR_SUCCESS;
 }
+
+/* https://github.com/anza-xyz/agave/blob/77daab497df191ef485a7ad36ed291c1874596e5/sdk/program/src/pubkey.rs#L578-L625 */
+int
+fd_pubkey_derive_pda( fd_pubkey_t const * program_id, 
+                      ulong               seeds_cnt, 
+                      uchar **            seeds, 
+                      uchar *             bump_seed, 
+                      fd_pubkey_t *       out ) {
+
+  if( seeds_cnt>MAX_SEEDS ) {
+    return FD_PUBKEY_ERR_MAX_SEED_LEN_EXCEEDED;
+  }
+  /* TODO: This does not contain size checks for the seed as checked in
+     https://github.com/anza-xyz/agave/blob/77daab497df191ef485a7ad36ed291c1874596e5/sdk/program/src/pubkey.rs#L586-L588 */
+                      
+  fd_sha256_t sha;
+  fd_sha256_init( &sha );
+  for ( ulong i=0UL; i<seeds_cnt; i++ ) {
+    uchar * seed = *(seeds + i);
+    if( !seed ) {
+      break;
+    }
+    
+    fd_sha256_append( &sha, seed, 32UL );
+    if( bump_seed ) {
+      fd_sha256_append( &sha, bump_seed, 1UL );
+    }
+    fd_sha256_append( &sha, program_id,              sizeof(fd_pubkey_t) );
+    fd_sha256_append( &sha, "ProgramDerivedAddress", 21UL                );
+
+    fd_sha256_fini( &sha, out );
+  }
```

Comment:
This iteration is wrong - we should hash all the seeds, then hash the bump seed, then hash the program id, then hash the marker. This loop hashes the bump seed, program id and marker every time.

See https://github.com/firedancer-io/firedancer/blob/vm-belt-sanding-public/src/flamenco/vm/syscall/fd_vm_syscall_pda.c#L12-L69